src/jdk.compiler/share/classes/com/sun/tools/javac/tree/TreeInfo.java

```diff
@@ -294,6 +315,9 @@ public static void setPolyKind(JCTree tree, PolyKind pkind) {
             case NEWCLASS:
                 ((JCNewClass)tree).polyKind = pkind;
                 break;
+            case DEFAULT_VALUE:
+                ((JCDefaultValue)tree).polyKind = pkind;
+                break;
             case REFERENCE:
```

Comment:
This is surprising. I would expected polyKind to be set in a similar fashion to switch and conditional expressions.